torch/nested/_internal/ops.py

```diff
@@ -937,7 +953,9 @@ def chunk_default(func, *args, **kwargs):
         # Note that the actual number of chunks returned is not necessarily the same as
         # the input number; it can be counter-intuitive, but it matches dense behavior.
         return [
-            NestedTensor(values=chunk_values[i], **(nested_kwargs[i]))
+            _construct_nested_tensor_compat(
```

Comment:
nit: I make a lot of changes to torch_dispatch impls of ops and I'd prefer to still be able to used the `NestedTensor` constructor so it feels less magical. can we alternatively define the helper at the level of a convenient way of constructing a metadata object that can then be passed to the usual constructor?
```python
nested_kwargs = [
    # helper for this CacheTensor construction?
    {"metadata": CacheTensor(per_offsets, {}), "_ragged_idx": inp._ragged_idx}
    for per_offsets in chunked_offsets
]
...
# same as before
return [
    NestedTensor(values=chunk_values[i], **(nested_kwargs[i]))
    ...
```

Reply:
We do have a way of constructing a metadata object :) 
Will rewrite to use that.